### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by RGologorsky** [mattiaspaul/OBELISK#5]
Thank you for sharing your code! I have a few questions on output size and ohem for the obelisk_visceral model. 

(1) for the obelisk_viscera model, why is the output half-resolution segmentation prediction and not full_res? Seems like one would want output to be the same size as the target segmentation mask, e.g. to make the loss func easier to calculate?

(2) how is the output converted to full-resolution (e.g. to calculate Cross Entropy loss)? In inference.py it seems like it's done by rescaling the input - "predict = F.interpolate(predict,size=[D_in0,H_in0,W_in0]". Yet if we need to rescale back to full-size, does it make sense to just return the full-size prediction in obelisk_visceral?

(3) Also I was wondering about the comment in obelisk_visceral def: 
"#pre-smooth image (has to be done in advance for original models )
#x00 = F.avg_pool3d(inputImg,3,padding=1,stride=1)"

Does this mean we should apply F.avg_pool3d(inputImg,3,padding=1,stride=1) to our input batches as a pre-preprocessing step? How does this compare to inference.py which does it 3 times? And why do we scale img_val by 500 specifically? In general, what smoothing/voxel intensity normalization or pre-processing do you recommend, if any?

if(modelname=='obelisk'):            
img_val = img_val/500.0            
img_val = F.avg_pool3d(img_val,5,stride=1,padding=2)            
img_val = F.avg_pool3d(img_val,5,stride=1,padding=2)            
img_val = F.avg_pool3d(img_val,3,stride=1,padding=1)           

 _,_,D_in1,H_in1,W_in1 = img_val.size()            
full_res = torch.Tensor([D_in1,H_in1,W_in1]).long()            

(4) Could you please explain what the ratio and weight parameters are for ohem and why they are set as they are in the code? Since OHEM selects the top instance losses to propagate backwards, could it be used in conjunction with any loss function (such as Dice), or is nll loss recommended (eg because of smoother loss landscape)?

class_weight = torch.sqrt(1.0/(torch.bincount(segs.view(-1)).float()))    
class_weight = class_weight/class_weight.mean()    
class_weight[0] = 0.5  
#criterion = nn.CrossEntropyLoss()#    
my_criterion = my_ohem(.25,class_weight.cuda())#0.25

Thanks!
